core/src/main/java/org/apache/iceberg/PartitionStats.java

```diff
@@ -249,4 +250,45 @@ public <T> void set(int pos, T value) {
         throw new UnsupportedOperationException("Unknown position: " + pos);
     }
   }
+
+  @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    } else if (!(other instanceof PartitionStats)) {
+      return false;
+    }
+
+    PartitionStats that = (PartitionStats) other;
+    return Objects.equals(partition, that.partition)
+        && specId == that.specId
+        && dataRecordCount == that.dataRecordCount
+        && dataFileCount == that.dataFileCount
+        && totalDataFileSizeInBytes == that.totalDataFileSizeInBytes
+        && positionDeleteRecordCount == that.positionDeleteRecordCount
+        && positionDeleteFileCount == that.positionDeleteFileCount
+        && equalityDeleteRecordCount == that.equalityDeleteRecordCount
+        && equalityDeleteFileCount == that.equalityDeleteFileCount
+        && totalRecordCount == that.totalRecordCount
+        && Objects.equals(lastUpdatedAt, that.lastUpdatedAt)
+        && Objects.equals(lastUpdatedSnapshotId, that.lastUpdatedSnapshotId);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        partition,
```

Comment:
Similar comment to above, StructLike doesn't have hash so we'll need to use JavaHashes.struct(...).hash(partition)

Reply:
Replied above.